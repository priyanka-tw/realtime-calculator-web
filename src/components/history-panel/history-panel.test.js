import React from 'react';
import {render} from '@testing-library/react';
import 'jest-styled-components';
import Context from "../../services/global-context-provider/context";
import WS from "jest-websocket-mock";
import HistoryPanel from "./history-panel";

describe('History panel', () => {
    let server, client;

    beforeEach(async () => {
        server = new WS("ws://localhost:8080/ws", {jsonProtocol: true});
        client = new WebSocket("ws://localhost:8080/ws");
    });

    afterEach(() => {
        WS.clean();
    });

    it('should render history panel with descending order of events', async () => {
        const {container} = render(<Context.Provider value={{socket: client}}>
            <HistoryPanel/> </Context.Provider>);
        await server.connected;

        server.send({event: 'history', data: '6*3 = 18'});
        server.send({event: 'history', data: '1+4 = 5'});

        expect(container).toMatchSnapshot();
    });

    it('should render history panel with maximum of 10 records', async () => {
        const {container} = render(<Context.Provider value={{socket: client}}>
            <HistoryPanel/> </Context.Provider>);
        await server.connected;

        server.send({event: 'history', data: '6*3 = 18'});
        server.send({event: 'history', data: '1+4 = 5'});
        server.send({event: 'history', data: '5-2 = 3'});
        server.send({event: 'history', data: '9*7 = 63'});
        server.send({event: 'history', data: '4/2 = 2'});
        server.send({event: 'history', data: '7-3 = 4'});
        server.send({event: 'history', data: '56*4 = 224'});
        server.send({event: 'history', data: '54-32 = 22'});
        server.send({event: 'history', data: '10+2 = 12'});
        server.send({event: 'history', data: '4*6 = 18'});
        server.send({event: 'history', data: '200*2 = 400'});

        expect(container).toMatchSnapshot();
    });

    it('should maintain order for newly entered result if existing values are same', async () => {
        const {container} = render(<Context.Provider value={{socket: client}}>
            <HistoryPanel/> </Context.Provider>);
        await server.connected;

        server.send({event: 'history', data: '6*3 = 18'});
        server.send({event: 'history', data: '6*3 = 18'});
        server.send({event: 'history', data: '6*3 = 18'});
        server.send({event: 'history', data: '6*3 = 18'});
        server.send({event: 'history', data: '6*3 = 18'});
        server.send({event: 'history', data: '6*3 = 18'});
        server.send({event: 'history', data: '6*3 = 18'});
        server.send({event: 'history', data: '6*3 = 18'});
        server.send({event: 'history', data: '6*3 = 18'});
        server.send({event: 'history', data: '6*5 = 30'});

        expect(container).toMatchSnapshot();
    });

});
