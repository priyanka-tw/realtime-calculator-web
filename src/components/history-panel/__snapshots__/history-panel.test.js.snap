// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`History panel should render history panel with maximum of 10 records 1`] = `
.c2 {
  width: 200px;
  margin: 10px auto;
  border: 3px solid black;
  padding: 10px;
  text-align: center;
  font-size: 14pt;
  color: black;
  font-family: verdana;
  border-radius: 25px;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
}

.c0 {
  color: red;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  height: -webkit-max-content;
  height: -moz-max-content;
  height: max-content;
  -webkit-flex: display;
  -ms-flex: display;
  flex: display;
  margin-left: 200px;
}

.c1 {
  font-size: 20px;
  font-family: verdana;
  color: black;
  text-align: center;
  margin-top: 45px;
}

<div>
  <div
    class="c0"
  >
    <h2
      class="c1"
    >
      History:
    </h2>
    <div>
      <div
        class="c2"
      >
        200*2 = 400
      </div>
      <div
        class="c2"
      >
        4*6 = 18
      </div>
      <div
        class="c2"
      >
        10+2 = 12
      </div>
      <div
        class="c2"
      >
        54-32 = 22
      </div>
      <div
        class="c2"
      >
        56*4 = 224
      </div>
      <div
        class="c2"
      >
        7-3 = 4
      </div>
      <div
        class="c2"
      >
        4/2 = 2
      </div>
      <div
        class="c2"
      >
        9*7 = 63
      </div>
      <div
        class="c2"
      >
        5-2 = 3
      </div>
      <div
        class="c2"
      >
        1+4 = 5
      </div>
    </div>
  </div>
   
</div>
`;
